provider/provider_upgrade_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016-2023, Pulumi Corporation.  All rights reserved.
+
+package gcp
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	pfbridge "github.com/pulumi/pulumi-terraform-bridge/pf/tfbridge"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+
+	"encoding/json"
+	ptest "github.com/pulumi/providertest"
+	"github.com/pulumi/pulumi-gcp/provider/v6/pkg/version"
+	"os"
+	"sort"
+	"strings"
+)
+
+func TestProviderUpgrade(t *testing.T) {
+	baseline := "6.67.0"
+
+	type testCase struct {
+		program string
+	}
+
+	testCases := []testCase{
+		// handcrafted
+		{"dns-recordset"},
+		{"pubsub-subscription"},
+		{"pubsub-topic"},
+		{"serviceaccount-account"},
+		{"storage-bucket"},
+		{"storage-bucketobject"},
+		{"secretmanager-secret"},
+		{"sql-user"},
+
+		// extracted from schema examples and manually corrected
+		{"bigquery-table"},
+		{"compute-firewall"},
+		{"cloudfunctions-function"},
+
+		// extracted as-is from schema examples
+		{"monitoring-alertpolicy-1"},
+		{"bigquery-datasetaccess-3"},
+		{"bigquery-routine-1"},
+		{"bigquery-routine-2"},
+		{"cloudrunv2-job-6"},
+		{"cloudrunv2-service-4"},
+		{"cloudrunv2-service-5"},
+		{"cloudrunv2-service-7"},
+		{"compute-disk-1"},
+		{"compute-disk-3"},
+	}
+
+	test := func(t *testing.T, tc testCase) {
+		ptest.VerifyUpgrade(t).
+			WithProviderName("gcp").
+			WithBaselineVersion(baseline).
+			WithProgram(filepath.Join("test-programs", tc.program)).
+			WithResourceProviderServer(providerServer(t)).
+			WithConfig("gcp:project", "pulumi-development").
+			Run()
+	}
```

Comment:
this is a totally insignificant nit, but why is this not just inline in the loop below?